File: example-base-27/java-interview-utilities-v4/src/main/java/com/prem/anagram/ReverseLinkedList.java
package com.prem.anagram;
public class ReverseLinkedList { public static class Node{public int data;public Node next;public Node(int d){data=d;}} public static Node reverse(Node head){Node prev=null,cur=head;while(cur!=null){Node next=cur.next;cur.next=prev;prev=cur;cur=next;}return prev;} }

File: example-base-29/python-algorithms-playground-v2/algorithms/linked_list/reverse_linked_list.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class Node:
    value: int
    next: Optional["Node"] = None

def reverse(head):
    previous = None
    current = head
    while current:
        nxt = current.next
        current.next = previous
        previous, current = current, nxt
    return previous

def to_list(head):
    result = []
    while head:
        result.append(head.value)
        head = head.next
    return result

if __name__ == "__main__":
    head = Node(1, Node(2, Node(3)))
    print(to_list(reverse(head)))
